mysite/lti/utils.py

```diff
@@ -16,3 +21,37 @@ def wrapped(request, *args, **kwargs):
         else:
             return fn(request, *args, **kwargs)
     return wrapped
+
+
+def create_courselets_user():
+    """Creates Courselets user w/ random username.
+
+    We can't trust LTI request w/o email in details.
+    """
+    password = str(uuid.uuid4())
+
+    created = False
+    while not created:
+        try:
+            username = generate_random_courselets_username()
+            with transaction.atomic():
+                courselets_user = User.objects.create_user(
+                    username=username,
+                    password=password,
+                )
+            created = True
+        except IntegrityError:
+            pass
+
+    return courselets_user
+
+
+def generate_random_courselets_username():
+    """
+    Create a valid random Courselets username.
+    """
+    allowable_chars = string.ascii_letters + string.digits
+    username = ''
+    for _index in range(30):
+        username = username + random.SystemRandom().choice(allowable_chars)
+    return username
```

Comment:
> Why not use some like uuid.uuid4().hex?

Django username has requirements - up to 30 characters.
We can get only 30 chars from 32 characters uuid.uuid4().hex may be. But I am not sure.